### PR TITLE
Improve JitBuilder testing

### PR DESCRIPTION
**Pull request by Leonardo2718**
These changes add some basic infrastructure for writing JitBuilder tests using Google Tests. Their is to simplify the process of developing JitBuilder tests, while also making them easier to read and understand. Two test cases are also included as initial examples of how to use the new infrastructure. 

Eventually, some of the examples in `jitbuilder/release/src` should be replaced by tests in `fvtest/jitbuildertests` because they only exist to test certain corner cases in JitBuilder.

These changes are a starting point for improving JitBuilder testing.

### CODE REVIEW

**Comment by Leonardo2718** [comment]
Once this gets merged, I will start opening issues to transform some of the JitBuilder examples into tests and to create new tests as well.

**Comment by mgaudet** [comment]
:tada: Awesome set of bones to build on. 

A couple of comments (I will review as well): 

- d988d63 typo in commit name (`JitBuilde`)

**Comment by mgaudet** [comment]
Lies. Don't really have much that I'm unhappy with in this PR: LGTM.

File: fvtest/jitbuildertest/JBTestUtil.hpp [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+ */
+#define DECL_TEST_BUILDER(name) \
+   class name : public TR::MethodBuilder { \
+      public: \
+      name(TR::TypeDictionary *); \
+      virtual bool buildIL(); \
+   }
+
+#define DEF_TEST_BUILDER_CTOR(name) \
+   inline name::name(TR::TypeDictionary *types) : TR::MethodBuilder(types)
```
Comment (mstoodle) [inline fvtest/jitbuildertest/JBTestUtil.hpp]: why inline?

File: fvtest/jitbuildertest/JBTestUtil.hpp [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+#define DEF_BUILDIL(name) \
+   inline bool name::buildIL()
```
Comment (mstoodle) [inline fvtest/jitbuildertest/JBTestUtil.hpp]: why inline?

File: fvtest/jitbuildertest/UnionTest.cpp [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+   DefineReturnType(Int32);
+   }
+
+bool
+TypePunInt32Int32Builder::buildIL()
+   {
+   StoreIndirect("TestUnionInt32Int32", "f1", Load("u"), Load("v1"));
+   StoreIndirect("TestUnionInt32Int32", "f2", Load("u"), Load("v2"));
+   Return(
+          LoadIndirect("TestUnionInt32Int32", "f1", Load("u")));
```
Comment (mstoodle) [inline fvtest/jitbuildertest/UnionTest.cpp]: not a huge deal, but why LoadIndirect on its own line but not Load? seems inconsistent

File: fvtest/jitbuildertest/selftest.cpp [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+bool
+BadBuilder::buildIL()
+   {
+   return false;
+   }
+
+class selftest : public JitBuilderTest {};
+
+/*
+ * `JustReturnTest` compiles the `JustReturn` method executes it. The test
```
Comment (mstoodle) [inline fvtest/jitbuildertest/selftest.cpp]: typo: method *and* executes it.

File: fvtest/jitbuildertest/JBTestUtil.hpp (hunk 2 above)
Comment (Leonardo2718) [inline fvtest/jitbuildertest/JBTestUtil.hpp]: Same reason as the other.

File: fvtest/jitbuildertest/JBTestUtil.hpp (hunk 1 above)
Comment (Leonardo2718) [inline fvtest/jitbuildertest/JBTestUtil.hpp]: It's inline to avoid creating duplicate symbols in the (unlikely) event that the constructor is defined in a header file that gets included in multiple translation units.

File: fvtest/jitbuildertest/UnionTest.cpp (hunk 3 above)
Comment (Leonardo2718) [inline fvtest/jitbuildertest/UnionTest.cpp]: fixed.

File: fvtest/jitbuildertest/selftest.cpp (hunk 4 above)
Comment (Leonardo2718) [inline fvtest/jitbuildertest/selftest.cpp]: fixed.

File: fvtest/jitbuildertest/JBTestUtil.hpp (hunk 1 above)
Comment (mstoodle) [inline fvtest/jitbuildertest/JBTestUtil.hpp]: Ok, good forward thinking there :) .

**Comment by mstoodle** [comment]
Thanks @Leonardo2718 , LGTM. I will merge once the testing completes.
